### PR TITLE
Seerat/hole tests

### PR DESCRIPTION
**Pull request by SeeratSek**
added tests for holes.go
removed what I believe was an extraneous function

### CODE REVIEW

File: server/arena/arena.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func (a *Arena) UpdatePositions() {
 	for _, hole := range a.Holes {
 		hole.Update()
 		if hole.Life < 0 {
-			hole.StartNewLife(a.generateCoordinate(models.MaxHoleRadius))
+			*hole = models.CreateHole(a.generateCoordinate(models.MaxHoleRadius))
```
Comment (brian-nguyen) [inline server/arena/arena.go]: I think the original intent here was to not have to reallocate a new instance of hole and instead "revive" the dead one by just assigning its attributes and only generating a new position. Either way works. I prefer this change.

File: server/models/hole.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ func CreateHole(position Position) Hole {
 	}
 }
```
Comment (brian-nguyen) [inline server/models/hole.go]: Let's change this to return a pointer to a Hole to avoid doing *hole = createHole

File: server/models/hole_test.go [hunk 3]
```diff
@@ -1,7 +1,69 @@
...
+	h := CreateHole(p)
+	h.Radius = MaxHoleRadius * 1.2
+	h.GravityRadius = 5
+	h.Update()
+	if h.Radius != (MaxHoleRadius * 1.2) {
+		t.Error("Radius increased over the max size")
+	}
+	if h.GravityRadius != 5 {
+		t.Error("Radius is incorrectly increased")
+	}
```
Comment (brian-nguyen) [inline server/models/hole_test.go]: I think this test can be a bit more sophisticated in terms of how we test that a hole can reach its max size. I think we can have a few test cases with different starting lives for each hole, and update them the expected amount of times until they're supposed to die and verify that they do die.

Comment (brian-nguyen) [reply server/models/hole_test.go]: Oh I see that this is supposed to be an edge case for the previous test, so this should be included in that same test as a subtest.

A test needs to be added to ensures the lifecycle of the hole is working.
